src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/FluidPump.java

```diff
@@ -153,6 +162,26 @@ protected void tick(@Nonnull Block b) {
                         nextFluid.setType(Material.AIR);
                     }
 
+                    return;
+                }
+                if (SlimefunUtils.isItemSimilar(menu.getItemInSlot(slot), emptyBottle, true, false)) {
```

Comment:
Semantically, an item cannot be an empty bucket AND and empty bottle.
You should probably also store `menu.getItemInSlot(slot)` as a variable perhaps
```suggestion
                } else if (SlimefunUtils.isItemSimilar(menu.getItemInSlot(slot), emptyBottle, true, false)) {
```